tests/resources/functional/snowflake/core_engine/lca/lca_homonym.sql

```diff
@@ -0,0 +1,24 @@
+-- snowflake sql:
+select ca_zip
+     from (
+      SELECT
+      substr(ca_zip,1,5) ca_zip,
+      trim(name) as name,
+      count(*) over( partition by ca_zip)
+      FROM customer_address
+      WHERE ca_zip IN ('89436', '30868'));
+-- databricks sql:
+SELECT
+  ca_zip
+FROM
+SELECT
+  SUBSTR(ca_zip,1,5) AS ca_zip,
+  TRIM(name) AS name,
+  COUNT(*) OVER (
+    PARTITION BY
+      SUBSTR(ca_zip,1,5)
+  )
+FROM
+  customer_address
+WHERE
+  SUBSTR(ca_zip,1,5) IN ('89436', '30868');
```

Comment:
This seems to be syntactically incorrect on Databricks (probably missing parenthesis around the subquery)

Reply:
The parenthesis issue is fixed in PR #1232.
This test is about homonyms, not subqueries, so I'm afraid I disagree with your comment re names.